### PR TITLE
show pie charts legend according to "hideLegend" field

### PR DESCRIPTION
**Pull request by davidbinyamin**
fixes https://github.com/demisto/etc/issues/37630

Report builder:
image

report:
image

### CODE REVIEW

File: src/components/Sections/SectionChart/SectionPieChart.js [hunk 1]
```diff
@@ -129,22 +129,24 @@ const SectionPieChart = ({ data, style, dimensions, legend, chartProperties = {}
...
-                    capitalize={legendStyle.capitalize === undefined || legendStyle.capitalize}
-                    data={preparedData}
-                    formatter={formatValue}
-                    height={legendHeight}
-                    style={legendStyle && legendStyle.style}
-                  />
-                }
-                {...legendStyle}
-              />
+              {preparedData.length > 1 && legendStyle && !legendStyle.hideLegend &&
```
Comment (yanyo87) [inline src/components/Sections/SectionChart/SectionPieChart.js:132]: preparedData.length > 0

**Comment by davidbinyamin** [comment]
@yanyo87 
Regarding the failed test... it is also failed on master locally 
image

File: src/components/Sections/SectionChart/SectionPieChart.js [hunk 2]
```diff
@@ -129,22 +129,24 @@ const SectionPieChart = ({ data, style, dimensions, legend, chartProperties = {}
...
-                    capitalize={legendStyle.capitalize === undefined || legendStyle.capitalize}
-                    data={preparedData}
-                    formatter={formatValue}
-                    height={legendHeight}
-                    style={legendStyle && legendStyle.style}
-                  />
-                }
-                {...legendStyle}
-              />
+              {preparedData.length > 0 && legendStyle && !legendStyle.hideLegend &&
```
Comment (yanyo87) [inline src/components/Sections/SectionChart/SectionPieChart.js:132]: if we don't  have legendStyle
we will not render the legend, 
what about BC ? 
would it brake ?

Comment (davidbinyamin) [reply src/components/Sections/SectionChart/SectionPieChart.js:132]: legendStyle object is not new, so it should be there for older versions.
regarding the "hideLegend" Same question applied to bar/line chart, no?

Comment (yanyo87) [reply src/components/Sections/SectionChart/SectionPieChart.js:132]: the code in bar/line for legendStyle is from day one

**Review by yanyo87 — APPROVED** [review]
👍

**Pull request merged**